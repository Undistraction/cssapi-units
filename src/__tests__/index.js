import {
  appendUnit,
  unitIsRemOrEm,
  pxToRemOrEmValue,
  remOrEmToPxValue,
  elementsOfUnitedNumber,
  unitedDimensionToUnitlessPixelValue,
} from '../index';

// const validRemValues = ['0rem', '-5rem', '-5.5rem', '5rem', '5.5rem'];
// const validEmValues = ['0em', '-5em', '-5.5em', '5em', '5.5em'];

describe('appendUnit', () => {
  it('appends the supplied unit', () => {
    expect(appendUnit(10, 'px')).toEqual('10px');
  });
});

describe('unitIsRemOrEm', () => {
  for (const value of ['em', 'rem']) {
    it(`returns true when value supplied is '${value}'`, () => {
      expect(unitIsRemOrEm(value)).toBeTruthy();
    });
  }

  for (const value of [
    5,
    '5',
    'px',
    '',
    null,
    {},
    undefined,
    [],
    true,
    false,
    NaN,
  ]) {
    it(`returns false when the value suppleid is '${value}'`, () => {
      expect(unitIsRemOrEm(value)).toBeFalsy();
    });
  }
});

describe('pxToRemOrEmValue', () => {
  it('returns the correct value', () => {
    expect(pxToRemOrEmValue(16, 16)).toBe(1);
    expect(pxToRemOrEmValue(16, 10)).toBe(1.6);
    expect(pxToRemOrEmValue(10, 16)).toBe(0.625);
    expect(pxToRemOrEmValue(-16, 10)).toBe(-1.6);
  });
});

describe('remOrEmToPxValue', () => {
  it('returns the correct value', () => {
    expect(remOrEmToPxValue(1, 16)).toBe(16);
    expect(remOrEmToPxValue(2, 10)).toBe(20);
    expect(remOrEmToPxValue(0.8, 16)).toBe(12.8);
    expect(remOrEmToPxValue(-2, 10)).toBe(-20);
  });
});

describe('elementsOfUnitedNumber', () => {
  for (const value of [
    5,
    '5',
    'px',
    'px10',
    '',
    null,
    {},
    undefined,
    [],
    true,
    false,
    NaN,
  ]) {
    it("throws if the value isn't in the form {number}{unit}", () => {
      expect(() =>
        elementsOfUnitedNumber(value)
      ).toThrowErrorMatchingSnapshot();
    });
  }

  it('returns the separate numeric and unit parts', () => {
    expect(elementsOfUnitedNumber('5px')).toEqual([5, 'px']);
    expect(elementsOfUnitedNumber('-10rem')).toEqual([-10, 'rem']);
    expect(elementsOfUnitedNumber('0em')).toEqual([0, 'em']);
  });
});

describe('unitedDimensionToUnitlessPixelValue', () => {
  expect(unitedDimensionToUnitlessPixelValue('16px', 16)).toEqual(16);
  expect(unitedDimensionToUnitlessPixelValue('1.6rem', 10)).toEqual(16);
  expect(unitedDimensionToUnitlessPixelValue('2em', 10)).toEqual(20);
});
