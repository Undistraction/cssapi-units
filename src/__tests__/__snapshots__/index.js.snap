// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elementsOfUnitedNumber throws if the value isn't in the form {number}{unit} 1`] = `"Supplied value was not a number with a unit: '5'"`;

exports[`elementsOfUnitedNumber throws if the value isn't in the form {number}{unit} 2`] = `"Supplied value was not a number with a unit: '5'"`;

exports[`elementsOfUnitedNumber throws if the value isn't in the form {number}{unit} 3`] = `"Supplied value was not a number with a unit: 'px'"`;

exports[`elementsOfUnitedNumber throws if the value isn't in the form {number}{unit} 4`] = `"Supplied value was not a number with a unit: 'px10'"`;

exports[`elementsOfUnitedNumber throws if the value isn't in the form {number}{unit} 5`] = `"Supplied value was not a number with a unit: ''"`;

exports[`elementsOfUnitedNumber throws if the value isn't in the form {number}{unit} 6`] = `"Supplied value was not a number with a unit: 'null'"`;

exports[`elementsOfUnitedNumber throws if the value isn't in the form {number}{unit} 7`] = `"Supplied value was not a number with a unit: '[object Object]'"`;

exports[`elementsOfUnitedNumber throws if the value isn't in the form {number}{unit} 8`] = `"Supplied value was not a number with a unit: 'undefined'"`;

exports[`elementsOfUnitedNumber throws if the value isn't in the form {number}{unit} 9`] = `"Supplied value was not a number with a unit: ''"`;

exports[`elementsOfUnitedNumber throws if the value isn't in the form {number}{unit} 10`] = `"Supplied value was not a number with a unit: 'true'"`;

exports[`elementsOfUnitedNumber throws if the value isn't in the form {number}{unit} 11`] = `"Supplied value was not a number with a unit: 'false'"`;

exports[`elementsOfUnitedNumber throws if the value isn't in the form {number}{unit} 12`] = `"Supplied value was not a number with a unit: 'NaN'"`;
